packages/elements-material-ui/src/Table/TableBody.tsx

```diff
@@ -83,7 +86,16 @@ export const TableBody: FC<TableTBodyProps<any>> = <T extends object>(props: Tab
                 row={row}
                 renderExpandedComponent={renderExpandedComponent}
               />
-              {pagination === 'infinite-scroll' && index === Math.ceil(rows.length * 0.7) && (
+              {isInfiniteScroll && isFirstWaypoint && index === rows.length - 4 && (
+                <Waypoint
+                  onEnter={({ previousPosition }) => {
+                    if (!loading && previousPosition !== 'above') {
+                      onInfiniteScroll?.();
+                    }
+                  }}
+                />
+              )}
+              {isInfiniteScroll && !isFirstWaypoint && index === rows.length - 15 && (
```

Comment:
I don't quite get it

Reply:
First time we render waypoint with trigger for next fetch after 16 items, then we render it after 25 items, then after 45... 

First time we need to have it at the bottom because in another case waypoint will be triggered immediately.
It's implemented to fetch more before the user scrolls to the bottom. 

I'll create vars for the "magic" numbers to prevent confusion, you are right.